cmd/go-cacher/cacher.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2023 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// The go-cacher binary is a cacher helper program that cmd/go can use.
+package main
+
+import (
+	"context"
+	"flag"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+
+	"github.com/bradfitz/go-tool-cache/cacheproc"
+	"github.com/bradfitz/go-tool-cache/cachers"
+)
+
+const defaultCacheKey = "v1"
+
+var (
+	serverBase = flag.String("cache-server", "", "optional cache server HTTP prefix (scheme and authority only); should be low latency. empty means to not use one.")
+	verbose    = flag.Bool("verbose", false, "be verbose")
+)
+
+func getAwsConfigFromEnv() (*aws.Config, error) {
+	// read from env
+	awsRegion, awsRegionOk := os.LookupEnv("GOCACHE_AWS_REGION")
+	if !awsRegionOk {
+		return nil, nil
+	}
+	accessKey, accessKeyOk := os.LookupEnv("GOCACHE_AWS_ACCESS_KEY")
+	secretAccessKey, secretKeyOk := os.LookupEnv("GOCACHE_AWS_SECRET_ACCESS_KEY")
```

Comment:
Instead of using LookupEnv, I would use GetEnv and check the strings are non blank; LookupEnv would return it's defined if I run it with `GOCACHE_AWS_REGION=` which isn't sensible